d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperConnectionManager.java

```diff
@@ -302,6 +309,11 @@ public void notifyEvent(ZKPersistentConnection.Event event)
     }
   }
 
+  public interface ZooKeeperConnectionWatcher
+  {
+    void onConnected();
+  }
+
```

Comment:
A little confused by this part. What this used for ? I neither find any code in `container` side.

Reply:
it's called right n this file on line 301.

Reply:
Yeah. I mean what is the `onConnected` function used for ? As I haven't see any implementation for this function